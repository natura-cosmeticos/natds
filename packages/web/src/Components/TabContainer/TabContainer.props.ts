import { TabsProps } from '@material-ui/core/Tabs'

export type ITabContainerProps = TabsProps;
