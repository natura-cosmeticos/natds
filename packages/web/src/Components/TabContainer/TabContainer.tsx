import * as React from 'react'
import MaterialTabContainer from '@material-ui/core/Tabs'
import { ITabContainerProps } from './TabContainer.props'

export const TabContainer = React.forwardRef<HTMLButtonElement, ITabContainerProps>((
  props: ITabContainerProps, ref
) => <MaterialTabContainer {...props} ref={ref} />)

TabContainer.displayName = 'TabContainer'

export default TabContainer
