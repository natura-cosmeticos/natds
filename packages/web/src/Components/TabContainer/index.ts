import TabContainer from './TabContainer'

export { ITabContainerProps } from './TabContainer.props'
export { TabContainer }
export default TabContainer
