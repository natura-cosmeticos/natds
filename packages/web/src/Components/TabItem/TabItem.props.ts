import { TabProps } from '@material-ui/core/Tab'

export type ITabItemProps = TabProps
