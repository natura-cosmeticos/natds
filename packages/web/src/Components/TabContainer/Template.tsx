import { Story } from '@storybook/react'
import * as React from 'react'
import { ITabContainerProps } from './TabContainer.props'
import { TabContainer } from './TabContainer'
import { TabItem } from '../TabItem'
import {
  WithBrandsProductIconAndLabel,
  WithLoveIconAndLabel,
  WithMakeupIconAndLabel,
  WithOutletIconAndLabel
} from '../TabItem/TabItem.stories'

export const Template : Story<ITabContainerProps> = (args: ITabContainerProps) => {
  const initialTabIndex = 0

  const [currentTabIndex, setCurrentTabIndex] = React.useState<number>(initialTabIndex)

  // eslint-disable-next-line @typescript-eslint/ban-types
  const handleChange = (event: React.ChangeEvent<{}>, newValue: number) => {
    setCurrentTabIndex(newValue)
  }

  return (
  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
  // @ts-ignore
    <TabContainer {...args} onChange={handleChange} value={currentTabIndex}>
      <TabItem {...WithLoveIconAndLabel.args} />
      <TabItem {...WithOutletIconAndLabel.args} />
      <TabItem {...WithBrandsProductIconAndLabel.args} />
      <TabItem {...WithMakeupIconAndLabel.args} />
    </TabContainer>
  )
}

export default Template
