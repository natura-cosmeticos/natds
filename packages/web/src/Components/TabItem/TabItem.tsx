import * as React from 'react'
import MaterialTabItem from '@material-ui/core/Tab'
import { ITabItemProps } from './TabItem.props'

export { ITabItemProps } from './TabItem.props'

export const TabItem = React.forwardRef<HTMLDivElement, ITabItemProps>(
  (props: ITabItemProps, ref) => (
    <MaterialTabItem {...props} ref={ref} />
  )
)

TabItem.displayName = 'TabItem'

export default TabItem
