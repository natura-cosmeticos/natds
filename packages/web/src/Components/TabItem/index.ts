import TabItem from './TabItem'

export { ITabItemProps } from './TabItem.props'
export { TabItem }
export default TabItem
