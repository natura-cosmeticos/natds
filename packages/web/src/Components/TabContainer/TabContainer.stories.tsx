import { Story } from '@storybook/react'
import { TabContainer } from './TabContainer'
import { ITabContainerProps } from './TabContainer.props'
import { Template } from './Template'
import { TabItem } from '../TabItem'

export default {
  component: TabContainer,
  subcomponents: { TabItem },
  title: 'Components/Tabs/Tab Container'
}

export const Playground : Story<ITabContainerProps> = Template.bind({})
Playground.args = {
  centered: false,
  indicatorColor: 'secondary',
  scrollButtons: 'auto',
  textColor: 'inherit',
  variant: 'standard'
}

export const Standard : Story<ITabContainerProps> = Template.bind({})
Standard.args = {
  variant: 'standard'
}

/**
 * @todo add width=600
 */
export const Scrollable : Story<ITabContainerProps> = Template.bind({})
Scrollable.args = {
  variant: 'scrollable'
}

/**
 * @todo add width=1240
 */
export const FullWidth : Story<ITabContainerProps> = Template.bind({})
FullWidth.args = {
  variant: 'fullWidth'
}

export const InheritTextColor : Story<ITabContainerProps> = Template.bind({})
InheritTextColor.args = {
  textColor: 'inherit'
}

export const PrimaryTextColor : Story<ITabContainerProps> = Template.bind({})
PrimaryTextColor.args = {
  textColor: 'primary'
}

export const SecondaryTextColor : Story<ITabContainerProps> = Template.bind({})
SecondaryTextColor.args = {
  textColor: 'secondary'
}

export const SecondaryIndicatorColor : Story<ITabContainerProps> = Template.bind({})
SecondaryIndicatorColor.args = {
  indicatorColor: 'secondary'
}

export const PrimaryIndicatorColor : Story<ITabContainerProps> = Template.bind({})
PrimaryIndicatorColor.args = {
  indicatorColor: 'primary'
}

/**
 * @todo add width=1240
 */
export const Centered : Story<ITabContainerProps> = Template.bind({})
Centered.args = {
  centered: true
}

export const ForAppBar : Story<ITabContainerProps> = Template.bind({})
ForAppBar.args = {
  centered: false,
  scrollButtons: 'off',
  textColor: 'inherit',
  variant: 'fullWidth'
}
