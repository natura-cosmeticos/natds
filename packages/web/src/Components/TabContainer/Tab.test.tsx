import * as React from 'react'
import * as TestRenderer from 'react-test-renderer'
import { TabContainer } from './TabContainer'
import { TabItem } from '../TabItem'

describe('Tab components', () => {
  describe('rendering variants', () => {
    test('should match to snapshot - empty TabContainer', () => {
      const testRenderer = TestRenderer.create(<TabContainer value={false} />)

      expect(testRenderer).toMatchSnapshot('TabContainer empty snapshot')
    })

    test('should match to snapshot - with TabContainer textColor applied', () => {
      const testRenderer = TestRenderer.create(<TabContainer value={false} textColor="secondary">
        <TabItem label="Item One" />
        <TabItem label="Item Two" />
        <TabItem label="Item Three" />
      </TabContainer>)

      expect(testRenderer).toMatchSnapshot('TabContainer textColor applied snapshot')
    })

    test('should match to snapshot - with TabContainer indicatorColor applied', () => {
      const testRenderer = TestRenderer.create(<TabContainer value={false} indicatorColor="secondary">
        <TabItem label="Item One" />
        <TabItem label="Item Two" />
        <TabItem label="Item Three" />
      </TabContainer>)

      expect(testRenderer).toMatchSnapshot('TabContainer indicatorColor applied snapshot')
    })

    test('should match to snapshot - with TabContainer centered', () => {
      const testRenderer = TestRenderer.create(<TabContainer value={false} centered>
        <TabItem label="Item One" />
        <TabItem label="Item Two" />
        <TabItem label="Item Three" />
      </TabContainer>)

      expect(testRenderer).toMatchSnapshot('TabContainer indicatorColor centered snapshot')
    })

    test('should match to snapshot - with TabItem disabled', () => {
      const testRenderer = TestRenderer.create(<TabContainer value={false}>
        <TabItem label="Item One" disabled />
        <TabItem label="Item Two" disabled />
        <TabItem label="Item Three" disabled />
      </TabContainer>)

      expect(testRenderer).toMatchSnapshot('TabContainer indicatorColor disabled snapshot')
    })
  })
})
