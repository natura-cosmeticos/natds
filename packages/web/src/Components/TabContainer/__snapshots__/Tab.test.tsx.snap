// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tab components rendering variants should match to snapshot - empty TabContainer: TabContainer empty snapshot 1`] = `
<div
  className="MuiTabs-root"
>
  <div
    className="MuiTabs-scroller MuiTabs-fixed"
    onScroll={[Function]}
    style={
      Object {
        "marginBottom": null,
        "overflow": "hidden",
      }
    }
  >
    <div
      className="MuiTabs-flexContainer"
      onKeyDown={[Function]}
      role="tablist"
    />
  </div>
</div>
`;

exports[`Tab components rendering variants should match to snapshot - with TabContainer centered: TabContainer indicatorColor centered snapshot 1`] = `
<div
  className="MuiTabs-root"
>
  <div
    className="MuiTabs-scroller MuiTabs-fixed"
    onScroll={[Function]}
    style={
      Object {
        "marginBottom": null,
        "overflow": "hidden",
      }
    }
  >
    <div
      className="MuiTabs-flexContainer MuiTabs-centered"
      onKeyDown={[Function]}
      role="tablist"
    >
      <button
        aria-selected={false}
        className="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="tab"
        tabIndex={-1}
        type="button"
      >
        <span
          className="MuiTab-wrapper"
        >
          Item One
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
      <button
        aria-selected={false}
        className="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="tab"
        tabIndex={-1}
        type="button"
      >
        <span
          className="MuiTab-wrapper"
        >
          Item Two
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
      <button
        aria-selected={false}
        className="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="tab"
        tabIndex={-1}
        type="button"
      >
        <span
          className="MuiTab-wrapper"
        >
          Item Three
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
    </div>
    <span
      className="PrivateTabIndicator-root-1 PrivateTabIndicator-colorSecondary-3 MuiTabs-indicator"
      style={
        Object {
          "left": 0,
          "width": 0,
        }
      }
    />
  </div>
</div>
`;

exports[`Tab components rendering variants should match to snapshot - with TabContainer indicatorColor applied: TabContainer indicatorColor applied snapshot 1`] = `
<div
  className="MuiTabs-root"
>
  <div
    className="MuiTabs-scroller MuiTabs-fixed"
    onScroll={[Function]}
    style={
      Object {
        "marginBottom": null,
        "overflow": "hidden",
      }
    }
  >
    <div
      className="MuiTabs-flexContainer"
      onKeyDown={[Function]}
      role="tablist"
    >
      <button
        aria-selected={false}
        className="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="tab"
        tabIndex={-1}
        type="button"
      >
        <span
          className="MuiTab-wrapper"
        >
          Item One
        </span>
      </button>
      <button
        aria-selected={false}
        className="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="tab"
        tabIndex={-1}
        type="button"
      >
        <span
          className="MuiTab-wrapper"
        >
          Item Two
        </span>
      </button>
      <button
        aria-selected={false}
        className="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="tab"
        tabIndex={-1}
        type="button"
      >
        <span
          className="MuiTab-wrapper"
        >
          Item Three
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`Tab components rendering variants should match to snapshot - with TabContainer textColor applied: TabContainer textColor applied snapshot 1`] = `
<div
  className="MuiTabs-root"
>
  <div
    className="MuiTabs-scroller MuiTabs-fixed"
    onScroll={[Function]}
    style={
      Object {
        "marginBottom": null,
        "overflow": "hidden",
      }
    }
  >
    <div
      className="MuiTabs-flexContainer"
      onKeyDown={[Function]}
      role="tablist"
    >
      <button
        aria-selected={false}
        className="MuiButtonBase-root MuiTab-root MuiTab-textColorSecondary"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="tab"
        tabIndex={-1}
        type="button"
      >
        <span
          className="MuiTab-wrapper"
        >
          Item One
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
      <button
        aria-selected={false}
        className="MuiButtonBase-root MuiTab-root MuiTab-textColorSecondary"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="tab"
        tabIndex={-1}
        type="button"
      >
        <span
          className="MuiTab-wrapper"
        >
          Item Two
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
      <button
        aria-selected={false}
        className="MuiButtonBase-root MuiTab-root MuiTab-textColorSecondary"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="tab"
        tabIndex={-1}
        type="button"
      >
        <span
          className="MuiTab-wrapper"
        >
          Item Three
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
    </div>
    <span
      className="PrivateTabIndicator-root-1 PrivateTabIndicator-colorSecondary-3 MuiTabs-indicator"
      style={
        Object {
          "left": 0,
          "width": 0,
        }
      }
    />
  </div>
</div>
`;

exports[`Tab components rendering variants should match to snapshot - with TabItem disabled: TabContainer indicatorColor disabled snapshot 1`] = `
<div
  className="MuiTabs-root"
>
  <div
    className="MuiTabs-scroller MuiTabs-fixed"
    onScroll={[Function]}
    style={
      Object {
        "marginBottom": null,
        "overflow": "hidden",
      }
    }
  >
    <div
      className="MuiTabs-flexContainer"
      onKeyDown={[Function]}
      role="tablist"
    >
      <button
        aria-selected={false}
        className="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit Mui-disabled Mui-disabled"
        disabled={true}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="tab"
        tabIndex={-1}
        type="button"
      >
        <span
          className="MuiTab-wrapper"
        >
          Item One
        </span>
      </button>
      <button
        aria-selected={false}
        className="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit Mui-disabled Mui-disabled"
        disabled={true}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="tab"
        tabIndex={-1}
        type="button"
      >
        <span
          className="MuiTab-wrapper"
        >
          Item Two
        </span>
      </button>
      <button
        aria-selected={false}
        className="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit Mui-disabled Mui-disabled"
        disabled={true}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="tab"
        tabIndex={-1}
        type="button"
      >
        <span
          className="MuiTab-wrapper"
        >
          Item Three
        </span>
      </button>
    </div>
  </div>
</div>
`;
